File: longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {


        Integer[] bucket = new Integer[nums.length];

        for (int i = 0; i< nums.length ; i++){
            // bucket[i] = Integer.valueOf( nums[i] );
            bucket[i] = Integer.valueOf(nums[i]) ;
        }

        // FOR ALL ELEME
            // SKIP IF PREV EXISTS
            // ELSE CHECK HOW MANY CONSEQUTIVE EXISTS

        Set<Integer> set =  new HashSet<>(Arrays.asList(bucket)); // hashset constructor accepts a list type 
// converting it into a list is  a On operation 
// Arrays.asList is a O1 operation because it simply provides a wrapper to the array , its an object of type arrays.arraylist not the utils.arraylist . a list of array.arraylist cannot be extended by adding or deleting more records, we should convert it to utils. arraylist 

        int max = 0;
        int curr_max=0;

        for (Integer ele : set){
            curr_max =0;
           if (set.contains(ele-1)){ // if prev ele exists
                continue;// we will start checking from the first ele onwards

            }
            else{
                while(set.contains(ele)){
                    curr_max ++;
                    ele ++;
                    
                }
                max = Math.max(curr_max, max);
            }
        }
        return max;
    }
}

File: Arrays & Hashing/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:

        def search( ele, nums, ct):
            if ele in nums  :
                ct+=1 
                return search(ele-1, nums, ct)

            else:
                return ct

        def fw_search( ele, nums, ct):
            if ele in nums  :
                nums.remove(ele)
                ct+=1 
                return fw_search(ele+1, nums, ct)

            else:
                return ct

        nums = set(nums) # to eleminate duplicates
        bw_ct = 0
        fw_ct = 0
        curr_max = 0


        if len(nums)==0:
            min_ele = -9999999999
        else:
            min_ele = min(nums)
 

 
        while len(nums)!=0: 
                # seach all its -1 elements
            ele = nums.pop()

            bw_ct = search(ele-1,nums, 1)
            fw_ct = fw_search(ele+1,nums,0)
            curr_max = max(curr_max, bw_ct + fw_ct)


            min_ele = min_ele + bw_ct + fw_ct+1

        
        return curr_max